cloudinit/net/activators.py

```diff
@@ -17,12 +17,13 @@ class NoActivatorException(Exception):
     pass
 
 
-def _alter_interface(cmd, device_name) -> bool:
+def _alter_interface(cmd, device_name, fail_on_stderr=False) -> bool:
     LOG.debug("Attempting command %s for device %s", cmd, device_name)
     try:
-        (_out, err) = subp.subp(cmd)
-        if len(err):
+        err = subp.subp(cmd).stderr
+        if err and fail_on_stderr and "Failed to connect system bus:" in err:
```

Comment:
I might be a little pedantic here, but don't we want the LOG anytime we have stderr? Something like:
```python
try:
    err = subp.subp(cmd).stderr
    if err:
        LOG.warning("...")
        if fail_on_stderr and "Failed to connect system bus:" in err:
            return False
    ...
```
?